transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/SchedulerUtil.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.scheduling.jobscheduling;
+
+public class SchedulerUtil {
+
+  static final String ATTEMPT_NUMBER = "attemptNumber";
+
+  static final String BACKEND_NAME = "backendName";
+
+  static final String APPLICATION_BUNDLE_ID = "appBundleId";
+
+  static long getScheduleDelay(long backendTimeDiff, int delta, int numberOfAttempts) {
+    if (numberOfAttempts > 3) {
+      return 1000000000;
+    }
+    return Math.max((long) (Math.pow(delta / 1000, numberOfAttempts)) * 1000, backendTimeDiff);
```

Comment:
I don't think we want to do `delta ^ attempt`, instead it is common to do `(2^attempt)*delta`